### PR TITLE
Improve deployment docs

### PR DESCRIPTION
**Pull request by gongjason**
Mostly condensing the information into 1 doc with references to detailed sections. It's much easier to deploy Firezone now than when we first wrote these docs.

### CODE REVIEW

File: docs/docs/administer/upgrade.md [hunk 1]
```diff
@@ -25,6 +25,12 @@ To upgrade Firezone, follow these steps:
 Occasionally problems arise. If you hit any, please let us know by [filing an
 issue](https://github.com/firezone/firezone/issues/new/choose).
 
+## Upgrading from >= 0.3.11 to >= 0.3.17
```
Comment (jamilbk) [inline docs/docs/administer/upgrade.md]: * Did you mean `Upgrading from 0.3.x to >= 0.3.16` as `>= 0.3.11 to >= 0.3.17` is somewhat ambiguous?
* Still need to test this to be sure this is the case
* I think 0.3.16 introduced this

We also need to be very clear here (make it bold, warning label) that any OIDC users added pre-`0.3.16` will need to revoke their tokens for this functionality to work, or the OAuth app needs to be deleted and recreated in their provider's portal.

File: docs/docs/deploy/index.md [hunk 2]
```diff
@@ -9,10 +9,84 @@ description: >
...
+
 Start by checking if your environment is listed on
 [supported platforms]({% link docs/deploy/supported-platforms.md %}).
 A kernel upgrade may be required to ensure WireGuard® is available.
 
-\
-[Start Here: Supported Platforms]({%link docs/deploy/supported-platforms.md%}){:.btn.btn-purple}
+### Firewall settings
+
+Firezone requires ports `80/tcp`, `443/tcp`, and `51820/udp` to be open.
```
Comment (jamilbk) [inline docs/docs/deploy/index.md]: This sort of contradicts what we say on the security considerations page: https://docs.firezone.dev/docs/administer/security-considerations/

Also, I'd reword this to be clear that typically `80/tcp` or `443/tcp` need to be available, not both. We highly recommend `443/tcp` and by default will redirect http requests to be https.

Also these can change so should clarify "by default"

Also they'll need to make sure forwarding is enabled in their firewall -- see the debugging https://docs.firezone.dev/docs/administer/troubleshoot/#debugging-connectivity-issues

File: docs/docs/deploy/index.md [hunk 3]
```diff
@@ -9,10 +9,84 @@ description: >
...
+```bash
+bash <(curl -Ls https://github.com/firezone/firezone/raw/master/scripts/install.sh)
+```
+
+This will ask you a few questions regarding your install, download the latest
+release for your platform, then create an administrator user and print to the
+console instructions for logging in to the web UI.
+
+By default, the web UI can be reached at the IP or domain name of your server.
+You can re-generate the admin credentials using the
```
Comment (jamilbk) [inline docs/docs/deploy/index.md]: Typo -> `regenerate`

File: docs/docs/deploy/index.md [hunk 4]
```diff
@@ -9,10 +9,84 @@ description: >
...
+connect to the VPN server:
+
+* [Add Users]({%link docs/user-guides/add-users.md%}):
+Add users to grant them access to your network.
+* [Client Instructions]({%link docs/user-guides/client-instructions.md%}):
+Instructions to establish a VPN session.
+
+## Troubleshooting
+
+Check the [troubleshooting guide]({% link docs/administer/troubleshoot.md %}).
```
Comment (jamilbk) [inline docs/docs/deploy/index.md]: First, check our troubleshooting guide to see if your issue is covered there

**Review by jamilbk — CHANGES_REQUESTED** [review]
Some content clarifications. Looks good, let's keep improving the quality of these docs!

**Comment by lilibobear** [comment]
1. In "Prerequisites", the section "Security Group and Firewall Settings" would be clearer if moved to the very top. When installing Firezone for a non-production case, an error can be made by clicking "skip to install the server" because the security group and firewall settings can be overlooked. This leads to an unsuccessful install.
2. Run sudo yum update prior to running the kernel upgrade. 
3. After running the kernel upgrade, there was an error "wireguard kernel module found, but not loaded. Load it with a sudo modprobe wireguard." Loaded sudo modprobe wireguard prior to the automatic batch install.
4. It's helpful to show a screenshot of what the terminal looks like when Firezone is installed correctly and the subsequent web UI interface.
5. Possibly provide a doc on how to SSH into a remote server?

File: docs/docs/administer/upgrade.md (hunk 1 above)
Comment (gongjason) [reply docs/docs/administer/upgrade.md]: oops this is supposed to be WIP in the OIDC docs branch. Removed it here.

**Review by jamilbk — APPROVED** [review]

File: docs/docs/deploy/prerequisites.md [hunk 5]
```diff
@@ -34,8 +34,9 @@ to do so:
 
 ## Security Group and Firewall Settings
 
-By default, Firezone requires ports `80/tcp`, `443/tcp`, and `51820/udp` to be open.
-This can change based on what you've configured in the configuration file.
+By default, Firezone requires ports `80/tcp` or `443/tcp`, and `51820/udp` to be
+accessible. We highliy recommend using https over `443/tcp`.
```
Comment (jamilbk) [inline docs/docs/deploy/prerequisites.md]: Typo: `highliy`

Comment (jamilbk) [inline docs/docs/deploy/prerequisites.md]: I think we can drop the `80/tcp` requirement as it's not *required* by default. For clarification, this could be

> By default, Firezone requires ports `443/tcp` and `51820/udp` traffic to be accessible for HTTPS and WireGuard traffic respectively.

File: docs/docs/deploy/index.md [hunk 6]
```diff
@@ -5,14 +5,112 @@ nav_order: 2
...
 A kernel upgrade may be required to ensure WireGuard® is available.
 
-\
-[Start Here: Supported Platforms]({%link docs/deploy/supported-platforms.md%}){:.btn.btn-purple}
+### Security settings
+
+Ensure port forwarding is enabled on your firewall.
+The default Firezone configuration requires the following ports to be open:
+
+* `80/tcp` or `443/tcp`: To access the web UI. We highly recommend https over `443/tcp`.
```
Comment (jamilbk) [inline docs/docs/deploy/index.md]: I would drop `80/tcp` here since we require `443/tcp` by default (we redirect 80 to 443 by default). You could then drop the `recommend https over 443/tcp` part -- might just be me but it reads like `we highly recommend https instead of 443/tcp`

**Review by jamilbk — CHANGES_REQUESTED** [review]
Just a few more minor cleanups

**Review by jamilbk — DISMISSED** [review]

**Review by jamilbk — APPROVED** [review]
